### ISSUE TITLE
Create `Pipeline` for running Tekton Pipeline CI/CD

### ISSUE THREAD

**Issue by bobcatfish** [tektoncd/pipeline#539]
## Expected Behavior

We should be testing Pipelines with Pipelines.

We should have a Pipeline which does the following, in this order (this is graph with 2 roots):

```
            unit tests (#532)                    build a test cluster (#534)      
                      |                                     |
     report result to github (#535)                         |
                      |                                     |
                       \                                   /
                         \                               /
          end to end tests (#533)          yaml tests (#536)      
                          | .                     |       
        report result to github (#535)   report result to github (#535)   
                           \                      /
                             teardown test cluster  (* see additional info)
```

("end to end tests" and "yaml tests" both execute in parallel, both depend on "build a test cluster" AND the unit test reporting results)

Runs should be triggered by new PR creation.

## Actual Behavior

The CI/CD used by Tekton Pipelines ([triggered via prow jobs](https://github.com/knative/test-infra/blob/master/ci/prow/config.yaml#L466)) invokes scripts only.

## Additional Info

* Should use #538 to invoke the Pipeline.
* Not sure best way to teardown cluster; probably need some way to execute a "finally" case (see #27). For the purposes of this issue we should just make something work

**Comment by abayer** [comment]
An idea for the "teardown" task - have it be a shell script run in parallel to the two branches already defined, that periodically queries GitHub to see if the commit status has been set for the two branches, and when they are set, blows away the cluster?

**Comment by bobcatfish** [comment]
* These days dogfooding is in full swing and most issues are in https://github.com/tektoncd/plumbing
* This is the project we are using to track our current dogfooding efforts: https://github.com/orgs/tektoncd/projects/1
* [This document contains our dogfooding plans](https://docs.google.com/document/d/1Tf3tdIBwkN5kxMuyYMo802F-DyOdsl_bgh-78a9prWQ/edit#)
